samples/subsys/mgmt/updatehub/overlay-wifi.conf

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2020 O.S.Systems
+# SPDX -License-Identifier: Apache-2.0
+
+# Disable default configs
+CONFIG_NET_L2_ETHERNET=n
+
+CONFIG_NET_DHCPV4=n
+
+CONFIG_NET_CONFIG_SETTINGS=n
```

Comment:
BTW, if you disable DHCPv4 and static IP address settings in the overlay, the IP addresses need to be come from somewhere else. Who is setting the IP addresses to the network interface in this case?

Reply:
Uhmm, I've been using ESP shield, default uses internal DHCP client. Which make sense since it is an offload driver.

See below the differences:

**CONFIG_NET_DHCPV4=n**
**CONFIG_NET_CONFIG_SETTINGS=n**
```
*** Booting Zephyr OS build zephyr-v2.4.0-3-gc2a0b0f50bdb  ***
[00:00:00.005,000] <inf> mcuboot: Starting bootloader
[00:00:00.006,000] <inf> mcuboot: Primary image: magic=good, swap_type=0x2, copy_done=0x1, image_ok=0x1
[00:00:00.006,000] <inf> mcuboot: Scratch: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
[00:00:00.006,000] <inf> mcuboot: Boot source: none
[00:00:00.006,000] <inf> mcuboot: Swap type: none


uart:~$ *** Booting Zephyr OS build zephyr-v2.4.0-2239-gacd86316efd9  ***
[00:00:00.249,000] <inf> wifi_esp: Waiting for interface to come up
[00:00:00.622,000] <inf> wifi_esp: ESP Wi-Fi ready
[00:00:00.628,000] <inf> main: UpdateHub sample app started
[00:00:00.628,000] <inf> main: Confirming the boot image
[00:00:00.632,000] <inf> main: Network disconnected
[00:00:06.274,000] <inf> main: Network connected
[00:00:06.274,000] <inf> main: Starting UpdateHub polling mode
[00:00:06.274,000] <inf> updatehub: SHA-256 verification on download and from flash
uart:~$ net iface 

Interface 0x20000538 (IP Offload) [1]
=====================================
Link addr : XX:XX:XX:XX:XX:XX
MTU       : 2048
IPv4 unicast addresses (max 1):
        192.168.10.104 DHCP preferred infinite
IPv4 multicast addresses (max 1):
        <none>
IPv4 gateway : 192.168.10.1
IPv4 netmask : 255.255.255.0
uart:~$ 
```

**CONFIG_NET_DHCPV4=y**
**CONFIG_NET_CONFIG_SETTINGS=n**
```
*** Booting Zephyr OS build zephyr-v2.4.0-3-gc2a0b0f50bdb  ***
[00:00:00.005,000] <inf> mcuboot: Starting bootloader
[00:00:00.006,000] <inf> mcuboot: Primary image: magic=good, swap_type=0x2, copy_done=0x1, image_ok=0x1
[00:00:00.006,000] <inf> mcuboot: Scratch: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
[00:00:00.006,000] <inf> mcuboot: Boot source: none
[00:00:00.006,000] <inf> mcuboot: Swap type: none


uart:~$ *** Booting Zephyr OS build zephyr-v2.4.0-2239-gacd86316efd9  ***
[00:00:00.250,000] <inf> wifi_esp: Waiting for interface to come up
[00:00:00.623,000] <inf> wifi_esp: ESP Wi-Fi ready
[00:00:00.629,000] <inf> main: UpdateHub sample app started
[00:00:00.629,000] <inf> main: Confirming the boot image
[00:00:00.633,000] <inf> main: Network disconnected
[00:00:06.274,000] <inf> main: Network connected
[00:00:06.274,000] <inf> main: Starting UpdateHub polling mode
[00:00:06.274,000] <inf> updatehub: SHA-256 verification on download and from flash
[00:00:08.642,000] <err> updatehub: Could not receive data
uart:~$ net iface 

Interface 0x20000540 (IP Offload) [1]
=====================================
Link addr : XX:XX:XX:XX:XX:XX
MTU       : 2048
IPv4 unicast addresses (max 1):
        192.168.10.104 DHCP preferred infinite
IPv4 multicast addresses (max 1):
        <none>
IPv4 gateway : 192.168.10.1
IPv4 netmask : 255.255.255.0
DHCPv4 lease time : 0
DHCPv4 renew time : 0
DHCPv4 server     : 0.0.0.0
DHCPv4 requested  : 0.0.0.0
DHCPv4 state      : disabled
DHCPv4 attempts   : 0
uart:~$
```

**CONFIG_NET_DHCPV4=n**
**CONFIG_NET_CONFIG_SETTINGS=y**
**CONFIG_NET_CONFIG_NEED_IPV4=y**
**CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.168.10.104"**
**CONFIG_NET_CONFIG_MY_IPV4_GW="192.168.10.1"**
```
*** Booting Zephyr OS build zephyr-v2.4.0-3-gc2a0b0f50bdb  ***
[00:00:00.005,000] <inf> mcuboot: Starting bootloader
[00:00:00.006,000] <inf> mcuboot: Primary image: magic=good, swap_type=0x2, copy_done=0x1, image_ok=0x1
[00:00:00.006,000] <inf> mcuboot: Scratch: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
[00:00:00.006,000] <inf> mcuboot: Boot source: none
[00:00:00.006,000] <inf> mcuboot: Swap type: none


uart:~$ *** Booting Zephyr OS build zephyr-v2.4.0-2239-gacd86316efd9  ***
[00:00:00.249,000] <inf> wifi_esp: Waiting for interface to come up
[00:00:00.621,000] <inf> wifi_esp: ESP Wi-Fi ready
[00:00:00.628,000] <inf> main: UpdateHub sample app started
[00:00:00.628,000] <inf> main: Confirming the boot image
[00:00:00.632,000] <inf> main: Network connected
[00:00:00.632,000] <inf> main: Starting UpdateHub polling mode
[00:00:00.632,000] <inf> updatehub: SHA-256 verification on download and from flash
[00:00:07.592,000] <err> updatehub: Could not receive data
uart:~$ net iface 

Interface 0x20000550 (IP Offload) [1]
=====================================
Link addr : XX:XX:XX:XX:XX:XX
MTU       : 2048
IPv4 unicast addresses (max 1):
        192.168.10.104 manual preferred infinite
IPv4 multicast addresses (max 1):
        <none>
IPv4 gateway : 192.168.10.1
IPv4 netmask : 255.255.255.0
uart:~$ 
```

**CONFIG_NET_DHCPV4=y**
**CONFIG_NET_CONFIG_SETTINGS=y**
**CONFIG_NET_CONFIG_NEED_IPV4=y**
**CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.168.10.104"**
**CONFIG_NET_CONFIG_MY_IPV4_GW="192.168.10.1"**
```
*** Booting Zephyr OS build zephyr-v2.4.0-3-gc2a0b0f50bdb  ***
[00:00:00.005,000] <inf> mcuboot: Starting bootloader
[00:00:00.006,000] <inf> mcuboot: Primary image: magic=good, swap_type=0x2, copy_done=0x1, image_ok=0x1
[00:00:00.006,000] <inf> mcuboot: Scratch: magic=unset, swap_type=0x1, copy_done=0x3, image_ok=0x3
[00:00:00.006,000] <inf> mcuboot: Boot source: none
[00:00:00.006,000] <inf> mcuboot: Swap type: none


uart:~$ *** Booting Zephyr OS build zephyr-v2.4.0-2239-gacd86316efd9  ***
[00:00:00.246,000] <inf> wifi_esp: Waiting for interface to come up
[00:00:00.618,000] <inf> wifi_esp: ESP Wi-Fi ready
[00:00:00.624,000] <inf> main: UpdateHub sample app started
[00:00:00.624,000] <inf> main: Confirming the boot image
[00:00:00.629,000] <inf> main: Network connected
[00:00:00.629,000] <inf> main: Starting UpdateHub polling mode
[00:00:00.629,000] <inf> updatehub: SHA-256 verification on download and from flash
[00:00:07.584,000] <err> updatehub: Could not receive data
uart:~$ net iface 

Interface 0x20000558 (IP Offload) [1]
=====================================
Link addr : 5C:CF:7F:72:4D:A2
MTU       : 2048
IPv4 unicast addresses (max 1):
        192.168.10.104 overridable preferred infinite
IPv4 multicast addresses (max 1):
        <none>
IPv4 gateway : 192.168.10.1
IPv4 netmask : 255.255.255.0
DHCPv4 lease time : 0
DHCPv4 renew time : 0
DHCPv4 server     : 0.0.0.0
DHCPv4 requested  : 0.0.0.0
DHCPv4 state      : init
DHCPv4 attempts   : 0
uart:~$ 
```

Reply:
Ok, makes sense.